v2/bigtable-cdc-to-hbase/src/main/java/com/google/cloud/teleport/v2/templates/transforms/ConvertChangeStream.java

```diff
@@ -0,0 +1,287 @@
+
+import com.google.cloud.bigtable.data.v2.models.ChangeStreamMutation;
+import com.google.cloud.bigtable.data.v2.models.DeleteCells;
+import com.google.cloud.bigtable.data.v2.models.DeleteFamily;
+import com.google.cloud.bigtable.data.v2.models.Entry;
+import com.google.cloud.bigtable.data.v2.models.Range.TimestampRange;
+import com.google.cloud.bigtable.data.v2.models.SetCell;
+import com.google.protobuf.ByteString;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import org.apache.beam.sdk.metrics.Metrics;
+import org.apache.beam.sdk.transforms.DoFn;
+import org.apache.beam.sdk.transforms.PTransform;
+import org.apache.beam.sdk.transforms.ParDo;
+import org.apache.beam.sdk.values.KV;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.hadoop.hbase.Cell;
+import org.apache.hadoop.hbase.CellScanner;
+import org.apache.hadoop.hbase.CellUtil;
+import org.apache.hadoop.hbase.client.Delete;
+import org.apache.hadoop.hbase.client.Mutation;
+import org.apache.hadoop.hbase.client.Put;
+import org.apache.hadoop.hbase.client.RowMutations;
+import org.apache.hadoop.util.Time;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Converts Bigtable change stream RowMutations objects to their approximating HBase mutations. */
+public class ConvertChangeStream {
+  private static final Logger LOG = LoggerFactory.getLogger(ConvertChangeStream.class);
+
+  /** Creates change stream converter transformer. */
+  public static ConvertChangeStreamMutation convertChangeStreamMutation() {
+    return new ConvertChangeStreamMutation();
+  }
+
+  /**
+   * Change stream converter that converts change stream mutations into RowMutation objects. Input
+   * is Bigtable change stream KV.of(rowkey, changeStreamMutation), Output is KV.of(rowkey, hbase
+   * RowMutations object)
+   */
+  public static class ConvertChangeStreamMutation
+      extends PTransform<
+          PCollection<KV<ByteString, ChangeStreamMutation>>,
+          PCollection<KV<byte[], RowMutations>>> {
+
+    /**
+     * Call converter with this function with the necessary params to enable two-way replication
+     * logic.
+     *
+     * @param enabled whether two-way replication logic is enabled
+     * @param cbtQualifierInput
+     * @param hbaseQualifierInput
+     */
+    public ConvertChangeStreamMutation withTwoWayReplication(
+        boolean enabled, String cbtQualifierInput, String hbaseQualifierInput) {
+      return new ConvertChangeStreamMutation(enabled, cbtQualifierInput, hbaseQualifierInput);
+    }
+
+    public ConvertChangeStreamMutation() {}
+
+    private ConvertChangeStreamMutation(
+        boolean enabled, String cbtQualifierInput, String hbaseQualifierInput) {
+      if (enabled) {
+        checkArgument(cbtQualifierInput != null, "cbt qualifier cannot be null.");
+        checkArgument(hbaseQualifierInput != null, "hbase qualifier cannot be null.");
+      }
+      twoWayReplication = enabled;
```

Comment:
isEnabled?

Reply:
Changed signature to bidirectionalReplicationEnabled